File: python/second.py
import unittest

class Element:
    pass

class Collection:
    def __init__(self):
        self.elements = []
    
    def add(self, element=None, collection=None):
        if collection is not None:
            self.elements.extend(collection.elements)
        else:
            self.elements.append(element)


class TestCase(unittest.TestCase):

    def setUp(self):
        unittest.TestCase.setUp(self)

    def tearDown(self):
        unittest.TestCase.tearDown(self)

    def test(self):
        e = Element()
        c = Collection()
        c.add(e)
        c1 = Collection()
        c1.elements.append(Element())
        c.add(collection=c1)
        self.assertTrue(len(c.elements) == 2)

if __name__ == '__main__':
    unittest.main()

File: rust/src/second.rs
pub struct Element {
}

pub struct Collection {
    pub elements: Vec<Element>    
}

trait Addable<T> {
    fn add(& mut self, t: T);
}

impl Addable<Element> for Collection {
    fn add(& mut self, t: Element) {
        self.elements.push(t);
    }
}

impl Addable<Collection> for Collection {
    fn add(& mut self, t: Collection) {
        self.elements.extend(t.elements);
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    
    #[test]
    fn test() {
        let e = Element{};
        let mut c = Collection{ elements: Vec::new() };
        let c1 = Collection{ elements: Vec::new() };
        c.add(e);
        c.add(c1);
    }
}
